### ISSUE TITLE
Fix guide to explain remote database not created until api call

### ISSUE THREAD

**Issue by rhythnic** [pouchdb/pouchdb#4708]
I'm doing the PouchDB guide, and according to the guide, the remote database should be created after calling
```
var remoteDB = new PouchDB('http://localhost:5984/dbname)
```
Verifying that the DB was created fails; however, if you continue through the guide and invoke the
```
remoteDB.info()
```
method, the remote database is created at that point.

**Comment by daleharvey** [comment]
Yup we should reword http://pouchdb.com/guides/databases.html to make sure thats clear, thanks for reporting

**Comment by daedlock** [comment]
@daleharvey shouldn't we promote this to make the remote database created in the constructor instead of waiting until `db.info()`. What's the strategy behind waiting for `db.info()` to be called?

**Comment by daleharvey** [comment]
Hey @daedlock, no we specifically changed the http adapter to have a stateless constructor, the main reason is that we cant have consistent error handling semantics while also having a synchronous constructor, ie:

```
var db = new PouchDB();
// it could throw, but it would be the only function that throws an error
```

```
new PouchDB('name').then(function(db) { 
// would be fine, but var db = new PouchDB('name') would be impossible, a promise cannot resolve to itself
```

It also confuses error handling semantics when for example 

```
var db = new PouchDB('name');
db.replicate.to('http://example.com', {retry: true});
```

If we had a constructor that throws then for consistency this should throw, but it wouldnt be consistent with any other usage where it rejects a promise / fires an error event.

By firing error on the first use of the instance, users only ever need to handle errors on method calls in a consistent way (and that consistency benefits the replicator etc as well)

**Comment by nolanlawson** [comment]
Fixed by 57923d6c088f52adcc785f1dad693227fb209396
